### PR TITLE
Bump psycopg to 3.2.1

### PR DESCRIPTION
**Pull request by RichieB2B**
This is an alternative to #695 which also changes the connection strings and fixes the unit tests.

### CODE REVIEW

**Review by pboers1988 — APPROVED** [review]

File: orchestrator/settings.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ class AppSettings(BaseSettings):
     EXECUTOR: str = ExecutorType.THREADPOOL
     WORKFLOWS_SWAGGER_HOST: str = "localhost"
     WORKFLOWS_GUI_URI: str = "http://localhost:3000"
-    DATABASE_URI: PostgresDsn = "postgresql://nwa:nwa@localhost/orchestrator-core"  # type: ignore
```
Comment (pboers1988) [inline orchestrator/settings.py:51]: Any chance of making this change backwards compatible that will save a lot of people some pain.

**Review by pboers1988 — CHANGES_REQUESTED** [review]

File: orchestrator/settings.py (hunk 1 above)
Comment (RichieB2B) [reply orchestrator/settings.py:51]: The whole reason for this MR is to switch to `psycopg` v3 so we can't assume the `psycopg2` module is available at all. We could convert `postgresql://` connection strings to `postgresql+psycopg://` and perhaps `postgresql+psycopg2://` as well? Perhaps issuing a warning that the conversion happened and the `DATABASE_URI` should be fixed.

Comment (pboers1988) [reply orchestrator/settings.py:51]: Yes the conversion and the warning would be great, that gives some time for people to migrate

Comment (pboers1988) [reply orchestrator/settings.py:51]: thanks @RichieB2B! I ran some tests internally, it seems to work 👍🏼

**Review by pboers1988 — APPROVED** [review]
LGTM

**Pull request merged**